problem9.py

```diff
@@ -0,0 +1,52 @@
+"""
+Created on Tue Mar 20 10:16:53 2018
+@author: patricio
+
+Let's use the fact that we can express a Pythagorean triplet in terms of
+two numbers m,n using the following relationships.
+
+       a = m^2 - n^2
+       b = 2 * m * n
+       c  = m^2 + n^2
+because,
+       a^2 = m^4 + n^4 – 2*m^2*n^2
+       b^2 = 4*m^2*n^2
+       c^2 = m^4 + n^4 + 2*m^2*n^2
+
+By looping over m, n we can generate pythagorean triplets instead of trying to
+loop over the three numbers (a, b, c)
+
+"""
+
+def build_pythagorean_triplets(limit):
+    m = 2
+    c = 0
+    pythagorean_list = []
+    #limiting c will also in consequence limit a and b
+    while(c < limit):
+        for n in range(1,m):
+            a = m**2 - n**2;
+            b = 2*m*n;
+            c = m**2 + n**2;
+            if c > limit:
```

Comment:
Feels weird to have this condition when we have something similar in the while statement, so I might be able to change some lines of code so the while loop ending feels more natural.

Reply:
Understood! I fixed it!